firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -26,4 +47,99 @@
  */
 @RunWith(AndroidJUnit4.class)
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
-public class FirebaseInstallationsInstrumentedTest {}
+public class FirebaseInstallationsInstrumentedTest {
+  private FirebaseApp firebaseApp;
+  @Mock private FirebaseInstallationServiceClient backendClientReturnsOk;
+  @Mock private FirebaseInstallationServiceClient backendClientReturnsError;
+  private FiidCache actualCache;
+  @Mock private FiidCache cacheReturnsError;
+
+  @Before
+  public void setUp() throws FirebaseInstallationServiceException {
+    MockitoAnnotations.initMocks(this);
+    FirebaseApp.clearInstancesForTest();
+    firebaseApp =
+        FirebaseApp.initializeApp(
+            ApplicationProvider.getApplicationContext(),
+            new FirebaseOptions.Builder()
+                .setApplicationId("1:123456789:android:abcdef")
+                .setProjectId("project-id")
+                .setApiKey("api_key")
+                .build());
+    actualCache = new FiidCache(firebaseApp);
+    when(backendClientReturnsOk.createFirebaseInstallation(
+            anyString(), anyString(), anyString(), anyString()))
+        .thenReturn(
+            InstallationResponse.builder()
+                .setName("/projects/123456789/installations/fid")
```

Comment:
Please use a properly formatted FID.
Maybe not necessary right now, but I have seen many tests in the past that had to be updated because a simple format check was added somewhere.
Just use something like "fis-testing-is-awesome" or "cccccccccccccccccccccc" which are both valid FIDs.

Reply:
Done. Thanks.